repositories/src/main/scala/cromwell/pipeline/datastorage/dao/repository/DocumentRepository.scala

```diff
@@ -16,5 +17,8 @@ class DocumentRepository(collection: MongoCollection[Document]) {
     collection.replaceOne(equal(fieldName, name), document, ReplaceOptions().upsert(true)).toFuture()
 
   def getByParam(field: String, name: String): Future[Seq[Document]] =
-    collection.find(regex(field, name)).toFuture()
+    collection.find(and(regex(field, name), notEqual("active", false))).toFuture()
+
+  def deactivateByParam(field: String, name: String): Future[UpdateResult] =
+    collection.updateOne(regex(field, name), set("active", false)).toFuture()
```

Comment:
DocumentRepository has no knowledge about fields that documents might have.
Although it might seem strange, but it will allow us to reuse this Repository in other Services too if we need to. You can express `deactivateByParam` in terms of `updateOne` in `ProjectConfigurationService` just like it's done in `addConfiguration` method.

Reply:
fixed.